frontend/src/scenes/project-homepage/projectHomepageLogic.ts

```diff
@@ -0,0 +1,16 @@
+import { kea } from 'kea'
+import { dashboardLogic } from 'scenes/dashboard/dashboardLogic'
+import { teamLogic } from 'scenes/teamLogic'
+import { projectHomepageLogicType } from './projectHomepageLogicType'
+
+export const projectHomepageLogic = kea<projectHomepageLogicType>({
+    path: ['scenes', 'project-homepage', 'projectHomepageLogic'],
+    connect: {
+        values: [
+            teamLogic,
+            ['currentTeam'],
+            dashboardLogic({ id: teamLogic.values.currentTeam?.primary_dashboard }),
```

Comment:
It took me a bit to land on this pattern for passing the id prop into the dashboard logic.

It works, but I'm not positive that there aren't "gotchas" here. @mariusandra, I'm curious if this smells to you?

Reply:
This smells a bit :). The problem is if you change the primary dashboard, this will not automatically update. 

![2022-03-08 09 35 36](https://user-images.githubusercontent.com/53387/157198846-9c9abe32-9ddd-4aee-968b-73443d60a78d.gif)

I'd try passing the dashboard ID as a prop to `projectHomepageLogic`, using it as the `key`, and connecting via `connect: (props) => { logic: [dashboardLogic({id: props.defaultDashboardId})] }` (or `values`).

Something like:

```ts
interface ProjectHomepageLogicProps {
    defaultDashboardId: number | undefined
}

export const projectHomepageLogic = kea<projectHomepageLogicType>({
    path: ['scenes', 'project-homepage', 'projectHomepageLogic'],
    props: {} as ProjectHomepageLogicProps,
    key: (props) => props.defaultDashboardId ?? 'none',
    connect: (props: ProjectHomepageLogicProps) => ({
        values: [
            teamLogic,
            ['currentTeam'],
            props.defaultDashboardId
                ? dashboardLogic({ id: props.defaultDashboardId })
                : () => ({ dashboard: null }),
            ['dashboard'],
        ],
    }),
})
```

Then change:

```ts
export const scene: SceneExport = {
    component: ProjectHomepage,
    logic: projectHomepageLogic,
    paramsToProps: () => ({ defaultDashboardId: getAppContext()?.current_team?.primary_dashboard }),
}
```

I tried locally, but there's some error running kea-typegen, which is also there on master even after https://github.com/PostHog/posthog/pull/8906 🤔 .

Reply:
Another related TODO would be to change the app context whenever the team changes, or pass the ID some other way...

Reply:
Oh interesting TIL you could pass props into connect.

Reply:
Hmmm, so I tried the above, but I hit issues with the logic still not being updated.

It's really odd, the `paramsToProps` get called as expected, and the `connect` function has the correct dashboardId, but when I inspect the logic's connections, it isn't the right dashboardLogic.

I also tried a different approach here too where the `dashboardLogic` got a new prop `isPrimaryDashboard`. If this this prop was set, then it fetched the current team's primary dashboard, and used that instead of any ids passed into it. This allows the logic to have static props, so it would work with turbo mode. I got it working, but it added too many edge cases + if statements to the `dashboardLogic`, and that logic is already complicated enough


I think I've officially spent too long on this one, so I'm going to close it out and work on other things. Might circle back on it when I have more of the homepage done.

Also, [this PR ](https://github.com/PostHog/posthog/pull/8930)should alleviate some of the user facing issues here by just speeding up the endpoint.